src/basics.md

```diff
@@ -277,6 +279,79 @@ fn main() {
 }
 ```
 
+[ex-rand-passwd]: #ex-rand-passwd
+<a name="ex-rand-passwd"></a>
+## Create random passwords from a set of alphanumeric characters
+
+[![rand-badge]][rand] [![cat-os-badge]][cat-os]
+
+Randomly generates a string of given length ASCII characters in the range `A-Z, a-z, 0-9`, with [`gen_ascii_chars`].
+
+```rust
+extern crate rand;
+
+use rand::{thread_rng, Rng};
+
+fn main() {
+    let s: String = thread_rng().gen_ascii_chars().take(30).collect();
+    println!("{}", s);
+}
+```
+
+[ex-rand-choose]: #ex-rand-choose
+<a name="ex-rand-choose"></a>
+## Create random passwords from a set of user-defined characters
+
+[![rand-badge]][rand] [![cat-os-badge]][cat-os]
+
+Randomly generates a string of given length ASCII characters with custom user-defined bytestring, with [`choose`].
+
+```rust
+extern crate rand;
+
+use rand::{thread_rng, Rng};
+
+struct CustomGenerator<'a, R:'a> {
+    rng: &'a mut R,
+}
+
+impl<'a, R: Rng> Iterator for CustomGenerator<'a, R> {
+    type Item = char;
+
+    fn next(&mut self) -> Option<char> {
+        const CHARSET: &'static [u8] =
+            b"ABCDEFGHIJKLMNOPQRSTUVWXYZ\
+              abcdefghijklmnopqrstuvwxyz\
+              0123456789)(*&^%$#@!~";
+        Some(*self.rng.choose(CHARSET).unwrap() as char)
+    }
+}
+
+fn gen_rand_chars<'a>(&'a mut self) -> CustomGenerator<'a, Self> where Self: Sized {
```

Comment:
I guess that you are trying to emulate the `gen_ascii_chars` method here but it is not possible to extend the Rng in such way. All we can do is prepare a standard constructor that will take an rng and a charset.

Some along the lines of
```rust
impl<'a, R> CustomGenerator<'a, R> {
    fn new(rng: R, charset: &'a [u8]) -> Result<Self> {
        if charset.len() > 0 {
            Some(CustomGenerator {
                rng: rng,
                charset: charset,
            })
        } else {
                 Err(ErrorKind::InvalidCharset.into())
        }
    }
}
```